skyportal/handlers/api/user.py

```diff
@@ -19,50 +20,131 @@
     Stream,
 )
 
+from skyportal.model_util import role_acls, all_acl_ids
 
+log = make_log("api/user")
 env, cfg = load_env()
 
 
+def set_default_role(user, session):
+    '''
+    Set the default role for a user. The default role can be set in the config file.
+    This method does not commit the session, so the session needs to be commited after calling this method.
+    '''
+    if (
+        cfg['user.default_role'] is not None
+        and isinstance(cfg['user.default_role'], str)
+        and cfg['user.default_role'] in role_acls
+    ):
+        role = session.query(Role).filter(Role.id == cfg['user.default_role']).first()
+        if role is None:
+            log(
+                f"Invalid default_role configuration value: {cfg['user.default_role']} does not exist"
+            )
+        else:
+            session.add(UserRole(user_id=user.id, role_id=role.id))
+
+
+def set_default_acls(user, session):
+    '''
+    Set the default acls for a user. The default acls can be set in the config file.
+    This method does not commit the session, so the session needs to be commited after calling this method.
+    '''
+    if cfg['user.default_acls'] is not None:
+        for acl_id in cfg['user.default_acls']:
+            if acl_id not in all_acl_ids:
+                log(
+                    f"Invalid default_acls configuration value: {acl_id} does not exist"
+                )
```

Comment:
In all these cases why do we want to log this instead of returning an error?

Reply:
For role and acls i agree that we could raise an error. But for the groups i'm not so sure.
If you do that for example, lets say you add 'Program A' as the default group, if you run `make load_demo_data`, as users are being posted before the groups, you'll just get an error and won't be able to load anything.

Reply:
Maybe @guynir i can do something like: if the group doesnt exist, i check if there are groups in the db other than sitewide, if there is none (meaning that no groups have been loaded yet), we just log, but if groups have been loaded and the groups in the config don't exist, we raise an error ?

Reply:
Do you expect to run into this situation where the groups aren't loaded yet? It seems like this should never happen...

Reply:
What about the example i gave before with the demo data ? In the first place, that is why i used logs instead of raising an error, as i couldnt load users because of it.

Reply:
I see. Are the users loaded before the groups? That doesn't sound like the right order. Is there any reason we don't load groups first?

Reply:
Yes. We load the groups after users so when creating the groups we can pass user ids in the field that allows to set group admins at the same time that we create the group. 
Maybe we can create the groups, then the users, and then set the group users ?

Reply:
I don't know if that would be so simple to do. Maybe give the demo data explicit group lists for each user, so that the handler doesn't use the defaults at all?

Reply:
Good idea, if setting the group users separately can't be done, i can simply give them the Sitewide Group explicitly so the default group part of the code does not run.

Reply:
That seems like the easiest way to go. And in general if there are non-existent groups in the defaults it should return an error. Make sure to catch those errors and turn them into `self.error` on the handler.